persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/SoftIndexFileStore.java

```diff
@@ -190,7 +193,8 @@ public void start() {
                      return null;
                   }
                   return null;
-               }).blockingSubscribe();
+               }).to(RxJavaInterop.flowableToCompletionStage());
+         CompletionStages.join(stage);
```

Comment:
Obviously not for this PR, but this has got me thinking about how we should handle the starting and initialization of stores in an async manner. At the very least, this would enable multiple stores to be started in parallel.